--- teeworlds-0.5.0-src.orig/default.bam.orig	2009-01-15 16:44:01.000000000 +0500
+++ teeworlds-0.5.0-src/default.bam	2009-01-15 22:22:50.000000000 +0500
@@ -6,7 +6,6 @@
 config = NewConfig()
 config:Add(OptFindCompiler())
 config:Add(OptTestCompileC("stackprotector", "int main(){return 0;}", "-fstack-protector -fstack-protector-all"))
-config:Add(OptFindLibrary("zlib", "zlib.h", false))
 config:Add(SDL.OptFind("sdl", true))
 config:Finalize("config.bam")
 
@@ -144,22 +143,10 @@
 		settings.link.libs:Add("shell32")
 	end
 	
-	-- compile zlib if needed
-	if config.zlib.value == 1 then
 		settings.link.libs:Add("z")
-		if config.zlib.include_path then
-			settings.cc.includes:Add(config.zlib.include_path)
-		end
-		zlib = {}
-	else
-		zlib = Compile(settings, Collect("src/engine/external/zlib/*.c"))
-		settings.cc.includes:Add("src/engine/external/zlib")
-	end
+		settings.cc.includes:Add("/usr/include")
+		settings.link.libs:Add("pnglite")
 
-	-- build the small libraries
-	wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
-	pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
-	
 	-- build game components
 	engine_settings = settings:Copy()
 	server_settings = engine_settings:Copy()
@@ -173,8 +160,11 @@
             client_settings.link.frameworks:Add("Cocoa")
 		else
 			client_settings.link.libs:Add("X11")
+			client_settings.link.libs:Add("Xrandr")
 			client_settings.link.libs:Add("GL")
 			client_settings.link.libs:Add("GLU")
+			client_settings.link.libs:Add("z")
+			client_settings.link.libs:Add("wavpack")
 		end
 	elseif family == "windows" then
 		client_settings.link.libs:Add("opengl32")
@@ -209,16 +199,15 @@
 	tools = {}
 	for i,v in ipairs(tools_src) do
 		toolname = PathFilename(PathBase(v))
-		tools[i] = Link(settings, toolname, Compile(settings, v), engine, zlib)
+		tools[i] = Link(settings, toolname, Compile(settings, v), engine)
 	end
 	
 	-- build client, server, version server and master server
 	client_exe = Link(client_settings, "teeworlds", game_shared, game_client,
-		engine, client, game_editor, zlib, pnglite, wavpack,
-		client_link_other, client_osxlaunch)
+		engine, client, game_editor, client_link_other, client_osxlaunch)
 
 	server_exe = Link(server_settings, "teeworlds_srv", engine, server,
-		game_shared, game_server, zlib)
+		game_shared, game_server)
 
 	serverlaunch = {}
 	if platform == "macosx" then
@@ -226,10 +215,10 @@
 	end
 		
 	versionserver_exe = Link(server_settings, "versionsrv", versionserver,
-		engine, zlib)
+		engine)
 
 	masterserver_exe = Link(server_settings, "mastersrv", masterserver,
-		engine, zlib)
+		engine)
 
 	-- make targets
 	c = PseudoTarget("client".."_"..settings.config_name, client_exe, client_depends)
--- teeworlds-0.5.0-src.orig/src/engine/client/ec_gfx.c.orig	2009-01-15 18:01:49.000000000 +0500
+++ teeworlds-0.5.0-src/src/engine/client/ec_gfx.c	2009-01-15 18:02:00.000000000 +0500
@@ -18,7 +18,7 @@
 #endif
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 #include <engine/e_client_interface.h>
 #include <engine/e_engine.h>
--- teeworlds-0.5.0-src.orig/src/engine/client/ec_snd.c.orig	2009-01-15 17:57:22.000000000 +0500
+++ teeworlds-0.5.0-src/src/engine/client/ec_snd.c	2009-01-15 17:57:44.000000000 +0500
@@ -6,7 +6,7 @@
 
 #include "SDL.h"
 
-#include <engine/external/wavpack/wavpack.h>
+#include <wavpack/wavpack.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
--- teeworlds-0.5.0-src.orig/src/tools/dilate.c.orig	2009-01-15 18:04:34.000000000 +0500
+++ teeworlds-0.5.0-src/src/tools/dilate.c	2009-01-15 18:04:57.000000000 +0500
@@ -1,6 +1,6 @@
 /* copyright (c) 2007 magnus auvinen, see licence.txt for more info */
 
-#include "../engine/external/pnglite/pnglite.c"
+#include <pnglite.h>
 
 typedef struct pixel_t
 {
